src/nvim/strings.c

```diff
@@ -588,18 +589,30 @@ static varnumber_T tv_nr(typval_T *tvs, int *idxp)
 ///                  value.
 /// @param[in,out]  idxp  Index in a list. Will be incremented.
 ///
+/// @param[out]  tofree If "tofree" is NULL get_tv_string_chk() is used. Some
+///                     types (e.g. List) are not converted to a string. If
+///                     "tofree" is not NULL encode_tv2echo() is used. All
+///                     types are converted to a string with the same format as
+///                     ":echo". The caller must free "*tofree".
+///
 /// @return String value or NULL in case of error.
-static char *tv_str(typval_T *tvs, int *idxp)
-  FUNC_ATTR_NONNULL_ALL FUNC_ATTR_WARN_UNUSED_RESULT
+static char *tv_str(typval_T *tvs, int *idxp, char_u **tofree)
+  FUNC_ATTR_NONNULL_ARG(1) FUNC_ATTR_NONNULL_ARG(2)
```

Comment:
Keep FUNC_ATTR_NONNULL_ALL. Also these two may be merged (`FUNC_ATTR_NONNULL_ARG(1, 2)`), but this is not needed: see next comment.